### PR TITLE
CLI - worker list

### PR DESCRIPTION
**Pull request by justcoon**
fixes: #300

### CODE REVIEW

File: golem-cli/src/clients/worker.rs [hunk 1]
```diff
@@ -418,3 +462,91 @@ struct Log {
...
+                comparator: filter_comparator_from_str(comparator)?,
+                value: worker_status_from_str(value)?,
+            })),
+            "created_at" | "createdAt" => Ok(WorkerFilter::CreatedAt(WorkerCreatedAtFilter {
+                comparator: filter_comparator_from_str(comparator)?,
+                value: DateTime::parse_from_str(value, "%Y-%m-%dT%H:%M:%SZ")
+                    .map(|ts| ts.into())
+                    .map_err(|e| format!("Invalid filter value: {}", e))?,
+            })),
+            _ if arg.starts_with("env.") => {
```
Comment (afsalthaj) [inline golem-cli/src/clients/worker.rs:529]: Looks a bit off. Could you explain what's going on here

File: golem-cli/src/clients/worker.rs [hunk 2]
```diff
@@ -80,6 +83,14 @@ pub trait WorkerClient {
         name: WorkerName,
         template_id: RawTemplateId,
     ) -> Result<WorkerMetadata, GolemError>;
+    async fn find_metadata(
+        &self,
+        template_id: RawTemplateId,
+        filter: Option<Vec<String>>,
```
Comment (afsalthaj) [inline golem-cli/src/clients/worker.rs:89]: I would rather call this as Option<workrfilter>. I don't think the call site need to be overly simplified to just pass vec<string>

**Review by afsalthaj — CHANGES_REQUESTED** [review]

File: golem-cli/src/clients/worker.rs (hunk 2 above)
Comment (justcoon) [reply golem-cli/src/clients/worker.rs:94]: it is using this API https://github.com/golemcloud/golem/blob/main/openapi/golem-service.yaml#L63

and it is list of filters, in form `{property} {operator} {value}`, for example `name = worker-name`, we need to do some documentation ....

File: golem-cli/src/clients/worker.rs [hunk 3]
```diff
@@ -227,6 +243,59 @@ impl<C: golem_client::api::WorkerClient + Sync + Send> WorkerClient for WorkerCl
...
+        template_id: RawTemplateId,
+        filter: Option<WorkerFilter>,
+        cursor: Option<u64>,
+        count: Option<u64>,
+        precise: Option<bool>,
+    ) -> Result<WorkersMetadataResponse, GolemError> {
+        info!(
+            "Getting workers metadata for template: {}, filter: {}",
+            template_id.0,
+            filter.is_some()
```
Comment (nicoburniske) [inline golem-cli/src/clients/worker.rs:257]: Just curious, does filter not have a debug impl?

**Review by nicoburniske — APPROVED** [review]
LGTM

File: golem-cli/src/clients/worker.rs (hunk 3 above)
Comment (justcoon) [reply golem-cli/src/clients/worker.rs:257]: yes, it have

File: golem-cli/src/worker.rs [hunk 4]
```diff
@@ -168,6 +168,29 @@ pub enum WorkerSubcommand {
         #[arg(short, long)]
         worker_name: WorkerName,
     },
+    /// Retrieves metadata about an existing workers in a template
+    #[command()]
+    List {
+        /// The Golem template the workers to be retrieved belongs to
```
Comment (senia-psm) [inline golem-cli/src/worker.rs:174]: Doc comment does not work on `#[command(flatten)]`.

File: golem-cli/src/worker.rs [hunk 5]
```diff
@@ -168,6 +168,29 @@ pub enum WorkerSubcommand {
...
+        #[command(flatten)]
+        template_id_or_name: TemplateIdOrName,
+
+        // Filter
```
Comment (senia-psm) [inline golem-cli/src/worker.rs:178]: It would be nice to have some explanation or example here.

File: golem-cli/src/worker.rs [hunk 6]
```diff
@@ -168,6 +168,29 @@ pub enum WorkerSubcommand {
...
+        #[arg(short, long)]
+        filter: Option<Vec<String>>,
+
+        // Cursor
```
Comment (senia-psm) [inline golem-cli/src/worker.rs:182]: Please add some details here (i.e. "skip this amount of records in result" or something)

File: golem-cli/src/worker.rs [hunk 7]
```diff
@@ -168,6 +168,29 @@ pub enum WorkerSubcommand {
...
+        #[arg(short, long)]
+        cursor: Option<u64>,
+
+        // Count
```
Comment (senia-psm) [inline golem-cli/src/worker.rs:186]: Please add some details here.

File: golem-cli/src/worker.rs [hunk 8]
```diff
@@ -168,6 +168,29 @@ pub enum WorkerSubcommand {
...
+        #[arg(short = 's', long)]
+        count: Option<u64>,
+
+        // Precise
```
Comment (senia-psm) [inline golem-cli/src/worker.rs:190]: Please add some details here.

**Pull request merged**